api/cloudcontroller/ccv2/errors_test.go

```diff
@@ -327,22 +347,65 @@ var _ = Describe("Error Wrapper", func() {
 
 			When("the error is a 5XX error", func() {
 				BeforeEach(func() {
-					serverResponseCode = http.StatusBadGateway
-					serverResponse = "I am some text"
+					serverResponseCode = http.StatusInternalServerError
+					serverResponse = "{}"
 				})
 
-				It("returns a V2UnexpectedResponseError with no json", func() {
-					_, _, err := client.GetApplications()
-					Expect(err).To(MatchError(ccerror.V2UnexpectedResponseError{
-						ResponseCode: http.StatusBadGateway,
-						V2ErrorResponse: ccerror.V2ErrorResponse{
-							Description: serverResponse,
-						},
-						RequestIDs: []string{
-							"6e0b4379-f5f7-4b2b-56b0-9ab7e96eed95",
-							"6e0b4379-f5f7-4b2b-56b0-9ab7e96eed95::7445d9db-c31e-410d-8dc5-9f79ec3fc26f",
-						},
-					}))
+				Context("(502) Bad Gateway", func() {
+					BeforeEach(func() {
+						serverResponseCode = http.StatusBadGateway
+					})
+
+					When("the service broker catalog is invalid", func() {
+						BeforeEach(func() {
+							requestMethod = http.MethodPost
+							requestPath = "/v2/service_brokers"
+							serverResponse = `{
+								"description": "Service broker catalog is invalid: \nService overview-service\n  Service dashboard client id must be unique\n",
+								"error_code": "CF-ServiceBrokerCatalogInvalid",
+								"code": 270012
+							}`
+						})
+
+						It("returns an ServiceBrokerCatalogInvalid", func() {
```

Comment:
Super nitpick (since we have other comments anyway): ServiceBrokerCatalogInvalid -> ServiceBrokerCatalogInvalidError